### PR TITLE
fix: custom webhook headers

### PR DESCRIPTION
**Pull request by JeevaRamu0104**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Enable custom webhook headers only if webhook url has some values.
image

## Checklist

- [ ] I ran `npm run re:build`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 1]
```diff
@@ -113,11 +119,19 @@ module WebHookAuthenticationHeaders = {
...
+    let outGoingWebhookDict =
+      formState.values
+      ->getDictFromJsonObject
+      ->getDictfromDict("outgoing_webhook_custom_http_headers")
     let outGoingWebhookArr =
       formState.values
       ->getDictFromJsonObject
       ->getDictfromDict("outgoing_webhook_custom_http_headers")
       ->Dict.keysToArray
+      ->Array.filterWithIndex((val, _) => {
```
Comment (sagarnaikjuspay) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:132]: can u use `Array.filter`

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 2]
```diff
@@ -113,11 +119,19 @@ module WebHookAuthenticationHeaders = {
...
+        outGoingWebhookDict->getOptionString(val)->Option.isSome ? true : false
```
Comment (sagarnaikjuspay) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:133]: `Option.isSome` return type itself bool

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 3]
```diff
@@ -185,22 +199,71 @@ module WebHookAuthenticationHeaders = {
...
+          <FieldRenderer
             field={DeveloperUtils.webhookUrl}
             labelClass="!text-base !text-grey-700 font-semibold"
             fieldWrapperClass="max-w-xl"
           />
         </div>
+        <div className="ml-4">
+          <div className={"mt-4 flex items-center text-jp-gray-700 font-bold self-start"}>
+            <div className="font-semibold text-base text-black dark:text-white">
+              {"Enbale Custom HTTP Headers"->React.string}
```
Comment (Riddhiagrawal001) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:249]: Typo`Enable`

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 4]
```diff
@@ -42,14 +42,20 @@ module AuthenticationInput = {
...
-    let outGoingWebhookDict =
-      formState.values
-      ->getDictFromJsonObject
-      ->getDictfromDict("outgoing_webhook_custom_http_headers")
+    let getOutGoingWebhook = () => {
+      let outGoingWebhookDict =
+        formState.values
+        ->getDictFromJsonObject
+        ->getDictfromDict("outgoing_webhook_custom_http_headers")
+      let key = outGoingWebhookDict->Dict.keysToArray->Array.at(index)->Option.getOr("")
```
Comment (Riddhiagrawal001) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:50]:
```suggestion
      let key = outGoingWebhookDict->Dict.keysToArray->LogicUtils.getValueFromArray(index,"")
```

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 5]
```diff
@@ -185,22 +199,71 @@ module WebHookAuthenticationHeaders = {
...
+      ReactFinalForm.useFormSubscription(["values"])->Nullable.make,
+    )
     let h2RegularTextStyle = `${HSwitchUtils.getTextClass((H3, Leading_1))}`
-
+    let webHookURL =
+      formState.values
+      ->getDictFromJsonObject
+      ->getString("webhook_url", "")
+      ->getNonEmptyString
+      ->Option.isSome
```
Comment (Riddhiagrawal001) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:214]:
```suggestion
      ->getOptionString("webhook_url")
      ->Option.isSome
```

**Review by Riddhiagrawal001 — DISMISSED** [review]

**Review by PritishBudhiraja — APPROVED** [review]

**Review by Riddhiagrawal001 — APPROVED** [review]

**Review by PritishBudhiraja — APPROVED** [review]

**Pull request merged**